dsn.go

```diff
@@ -13,73 +14,43 @@ type fireboltSettings struct {
 	accountName string
 }
 
-// splitString split string into two string, when first split char is encountered
-// return two strings:
-//    - first string before the encountered character, with removed backtick escape
-//    - second string is remaining string, including the character on which it was split
-func splitString(str string, splitChars []uint8) (string, string) {
-	var res string
-	for i := 0; i < len(str); i++ {
-		if str[i] == '\\' {
-			res += string(str[i+1])
-			i += 1
-			continue
-		}
-		for _, v := range splitChars {
-			if str[i] == v {
-				return res, str[i:]
-			}
-		}
-		res += string(str[i])
-	}
-	return res, ""
-}
-
-// parseRemainingDSN checks whether dsn has an expected prefix, removes it, and return two strings:
-//    - before the stopChars
-//    - after the stopChars, including the char itself
-func parseRemainingDSN(dsn string, expectedPrefix string, stopChars []uint8) (string, string, error) {
-	if !strings.HasPrefix(dsn, expectedPrefix) {
-		return "", dsn, fmt.Errorf("expected prefix not found: %s", expectedPrefix)
-	}
-
-	dsn = dsn[len(expectedPrefix):]
-	res, dsn := splitString(dsn, stopChars)
-	return res, dsn, nil
-}
+const dsnPattern = `^firebolt://(?P<username>.*@.*):(?P<password>.*)@(?P<database>\w+)(?:/(?P<engine>[^?]+))?(?:\?(?P<parameters>\w+\=[^=&]+(?:\&\w+=[^=&]+)*))?$`
```

Comment:
Nice one. Tried it out and it seems to work well. One thing - we need to make sure parameters don't contain `@` or parsing gets messed up.